python/taichi/lang/common_ops.py

```diff
@@ -0,0 +1,56 @@
+class TaichiOperations:
+    def __neg__(self):
+        import taichi as ti
+        return ti.neg(self)
+
+    def __abs__(self):
+        import taichi as ti
+        return ti.abs(self)
+
+    def __add__(self, other):
+        import taichi as ti
+        return ti.add(self, other)
+
+    __radd__ = __add__
+
+    def __sub__(self, other):
+        import taichi as ti
+        return ti.sub(self, other)
+
+    def __rsub__(self, other):
+        import taichi as ti
+        return ti.sub(other, self)
+
+    def __mul__(self, other):
+        import taichi as ti
+        return ti.mul(self, other)
+
+    __rmul__ = __mul__
```

Comment:
```suggestion
    def __rmul__(self, other):
        import taichi as ti
        return ti.mul(other, self)
```

Same here - think about cases like quaternion multiplications.